gwctl/go.mod

```diff
@@ -1,6 +1,6 @@
 module sigs.k8s.io/gateway-api/gwctl
 
-go 1.21
+go 1.22
```

Comment:
is the bump needed ?

Reply:
Using golang `slices.Concat` requires 1.22.

Reply:
ok reverted the bump to go1.21